### PR TITLE
HIVE-25139: Filter out null table properties in HiveIcebergMetaHook

### PR DESCRIPTION
**Pull request by lcspinter**

### CODE REVIEW

**Comment by lcspinter** [comment]
@marton-bod @pvary @szlta Could you please review this PR? Thanks

File: iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java [hunk 1]
```diff
@@ -297,6 +297,9 @@ private void updateHmsTableProperties(org.apache.hadoop.hive.metastore.api.Table
           "Table location not set");
     }
 
+    // Remove null values from hms table properties
+    hmsTable.getParameters().entrySet().removeIf(e -> e.getValue() == null);
```
Comment (marton-bod) [inline iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java:301]: Do we need to remove null keys here too, just like in `getCatalogProperties`?

Comment (lcspinter) [reply iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java]: Right, changed it.

File: iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java [hunk 2]
```diff
@@ -315,10 +318,10 @@ private void updateHmsTableProperties(org.apache.hadoop.hive.metastore.api.Table
   private static Properties getCatalogProperties(org.apache.hadoop.hive.metastore.api.Table hmsTable) {
     Properties properties = new Properties();
 
-    hmsTable.getParameters().forEach((key, value) -> {
+    hmsTable.getParameters().entrySet().stream().filter(e -> e.getKey() != null && e.getValue() != null).forEach(e -> {
       // translate key names between HMS and Iceberg where needed
-      String icebergKey = HiveTableOperations.translateToIcebergProp(key);
-      properties.put(icebergKey, value);
+      String icebergKey = HiveTableOperations.translateToIcebergProp(e.getKey());
```
Comment (pvary) [inline iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java:323]: nit: Why not
```
hmsTable.getParameters().entrySet().stream().filter(e -> e.getKey() != null && e.getValue() != null).forEach((key, value) -> {
```
and then we do not need these changes? Is it nicer or easier to read?

**Review by pvary — APPROVED** [review]

**Pull request merged**